### ISSUE TITLE
Image not regenerated when moving focal point

### ISSUE THREAD

**Issue by hgrimelid** [spatie/statamic-responsive-images#126]
Using a fresh installation with Statamic 3.3.9 and Statamic responive images 2.11.2.

I upload an image, then set focal point via asset editor and saving. 

When I reload the page with the image (included via the `responsive` tag), I expect the image to be updated with the new focal point, but that don't happen until I run `php please glide:clear`.  Is this intended behaviour, or is there something I can do to make the image get regenerated after saved/updated? My assets config is the default, with `'cache' => false,`

Using the Glide tag the image is regenerated as expected when reloading page.

**Comment by josephrlee** [comment]
I'm also running into this issue too. I have to clear the image cache to see the updated focal point-based crop.

**Comment by hgrimelid** [comment]
I created a new Statamic installation and have tested with different versions of Statamic.

It appears that the images are regenerated as expected in 3.3.7 and it stops working in 3.3.8.

Latest version `2.11.2` of `statamic-responsive-images` is used.

**Comment by riasvdv** [comment]
If you add `glide:fit="crop_focal"` to the responsive tag, does it work as expected then?

**Comment by hgrimelid** [comment]
I'm afraid not.

**Comment by taylorcammack** [comment]
Also experiencing this.

**Comment by stuartcusackie** [comment]
Me too. Have we figured out if it is a Statamic issue or if this package needs a PR?

**Comment by stuartcusackie** [comment]
I did a little bit of digging and I assume Statamic clears the cache [here](https://github.com/statamic/cms/blob/3a6b26116a6cf1609d1da116c0d626431bdb38fb/src/Assets/Asset.php#L489) or [here](https://github.com/statamic/cms/blob/3a6b26116a6cf1609d1da116c0d626431bdb38fb/src/Assets/AssetRepository.php#L128).

So why don't the files created by this package not get cleared? My best guess is that their urls are using a different format and cannot be found? I'm comparing two sites and noticing a lot of difference between the format of the urls:

**Spatie responsive Example url:**
http://spatsite.test/img/asset/aW1hZ2VzL01BWS0yNFRILU5FVy1ob21lcGFnZTItMTY1MzQ4MzI0My5wbmc=?w=267&bg=FDC5B9&h=190.87781350482&s=495fd2b93f5bb244a6521824d1f38433

**Statamic glide example url:**
http://statsite.test/img/assets/images/heroes/unsplash-justin-luebke.jpg?w=900&h=1200&fm=webp&s=02e08b1c1a9a84b81c59b3043c0951b6)

- Why is the url segment 'asset' instead of 'assets'?
- Why are the container name / subfolders / filename completely replaced with a random string?
- Should the height param be rounded?

**Comment by stuartcusackie** [comment]
I found a fix based on my last comment. It probably needs cleaning up but it works:

```
ResponsiveTag.php

class ResponsiveTag extends Tags
{
    protected static $handle = 'responsive';

    public static function render(...$arguments): string
    {
        $asset = $arguments[0];
        $parameters = $arguments[1] ?? [];

        // FIX: Enable automatic cropping
        if (Config::get('statamic.assets.auto_crop') && $asset) {
            $parameters['glide:fit'] = 'crop-'.$asset->get('focus', '50-50');
        }
       
       //.....

```

**Comment by ncla** [comment]
Correct me if I am wrong, but I think that PR https://github.com/statamic/cms/pull/5489 only currently relates to **static page cache** invalidation if an asset has changed. I downloaded the update and the issue is still there.

Regardless, I was ready to jump in and create a PR for this. stuartcusackie's comments have been helpful for my own code deep diving. From my understanding, and I agree with Stuart, we just need to have params parity with the way Statamic Glide tag itself generates these URLs. Take into account `statamic.assets.auto_crop` config setting on the way.

> And if I dump the $asset in various files I get different results:
> 
> * ResponsiveTag.php: focal points are correct
> * Breakpoint.php: focal points are always 50-50-1

I have stumbled upon this issue as well too, there is something weird going on with `Asset` metadata hydrating. See my comments here: https://github.com/spatie/statamic-responsive-images/issues/143#issuecomment-1172880518

> It's not ideal as I am re-retrieving each asset per breakpoint but it's the best I've got! I hope someone else can take it from here.

Don't feel too bad, hehe. Statamic CMS own `GlideImageGenerator` does this here for `fit` param. https://github.com/statamic/cms/blob/b719542205c7848fc13894ff002b1dc3051f9eee/src/Imaging/GlideImageManipulator.php#L208-L227
